src/urls/util.ts

```diff
@@ -53,8 +30,11 @@ function isBowerDependencyUrl(htmlUrl: OriginalDocumentUrl): boolean {
 /**
  * Rewrite a url to replace a `.js` file extension with `.html`.
  */
-function fixHtmlExtension(htmlUrl: string): string {
-  return htmlUrl.substring(0, htmlUrl.length - htmlExtension.length) + '.js';
+function fixHtmlExtensionIfFound(htmlUrl: string): string {
```

Comment:
Relatively minor, but IMO a very short function that's called only once is a candidate for inlining. Documenting the behavior and invariants of the function, and using that knowledge when reasoning around the call site is more cost than benefit.

Reply:
I agree, there's already a comment at the call site so it's not like there's much of a clarity win either. Moving to inline...

Reply:
I disagree a little with this. There's a benefit from abstracting away both the operation and the cast to a new interface. There's also a benefit to abstracting _all_ path/URL conversions, not just the ones that happen more than once.

I think the callsite comment can go away with a good name for the function. "fix" probably isn't the right verb here, since the URL isn't broken, it's just referring to a file that's being converted to JS.

Reply:
There's benefit, and there's cost. Moving the code into a function creates an abstraction, and in this case it makes the code harder to read and reason about. The benefit to the abstraction seems nebulous and pings my YAGNI-senses. Maybe we'll want to change and audit all path/url conversions, but probably not.

If we went whole hog and used branded string types for the different kinds of URLs here I think it would tip the scales for me, as then the abstraction would pay us back with type errors and find references and such.

Reply:
Given that this was a 1 time, 1-4 line operation, inlining it makes the most sense to me. There's always a time and place for simple operations to get their own helpers, but I'd only explicitly ask to see one for something very tricky or gross or complex (more docs than code?).

Regardless, now this is needed in a second place (see discussion in #219) so it's a bit of a moot point now.